test(linter/plugins): automate testing all fixtures

overlookmotel opened the pull request:
Remove the need to list every fixture in `e2e.test.ts` file by automatically running tests on every fixture in `test/fixtures` directory.

Running Oxlint with `--fix` and running fixtures with ESLint is now controlled by adding an `options.json` file to the fixture directory contining e.g. `{ "fix": true }` or `{ "eslint": true }`.

Now we can add a new fixture just by creating a new directory in `fixtures`.

overlookmotel commented on apps/oxlint/test/utils.ts:
```diff
@@ -35,28 +91,52 @@ interface TestFixtureOptions {
+        snapshot += `\n# File altered: ${filename}\n\`\`\`\n${codeAfter}\n\`\`\`\n`;
+      }
+    }),
+  );
```
JS is single-threaded, so what you described can't happen.

However, I had omitted to sort the results before adding to snapshot, which is required because the order mismatches are found in is unpredictable due to the parallelism.

Have fixed that now.

camc314 commented:
### Merge activity

* **Nov 21, 11:37 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 21, 11:37 AM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

overlookmotel committed:
test(linter/plugins): automate testing all fixtures (#15911)

Remove the need to list every fixture in `e2e.test.ts` file by automatically running tests on every fixture in `test/fixtures` directory.

Running Oxlint with `--fix` and running fixtures with ESLint is now controlled by adding an `options.json` file to the fixture directory contining e.g. `{ "fix": true }` or `{ "eslint": true }`.

Now we can add a new fixture just by creating a new directory in `fixtures`.
